assets/js/googlesitekit/datastore/user/surveys.test.js

```diff
@@ -0,0 +1,173 @@
+import { STORE_NAME } from './constants';
+
+describe( 'core/user surveys', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	const survey = {
+		survey_payload: 'foo',
+		session: 'bar',
+	};
+	const surveyTriggerEndpoint = /^\/google-site-kit\/v1\/core\/user\/data\/survey-trigger/;
+	const surveyEventEndpoint = /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/;
+
+	describe( 'actions', () => {
+		describe( 'triggerSurvey', () => {
+			it( 'throws an error when parameters are missing or incorrect', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey();
+				} ).toThrow( 'triggerID is required and must be a string' );
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey( 'coolSurvey', false );
+				} ).toThrow( 'options must be an object' );
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey( 'warmSurvey', { ttl: 'a' } );
+				} ).toThrow( 'options.ttl must be a number' );
+			} );
+
+			it( 'does not throw an error when parameters are correct', () => {
+				muteFetch( surveyTriggerEndpoint, [] );
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey( 'adSenseSurvey' );
+				} ).not.toThrow();
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).triggerSurvey( 'analyticsSurvey', { ttl: 1 } );
+				} ).not.toThrow();
+			} );
+
+			it( 'makes network requests to endpoints', async () => {
+				muteFetch( surveyTriggerEndpoint, [] );
+
+				await act( () => registry.dispatch( STORE_NAME ).triggerSurvey( 'optimizeSurvey', { ttl: 1 } ) );
+
+				expect( fetchMock ).toHaveFetched( surveyTriggerEndpoint, {
+					body: {
+						data: { triggerID: 'optimizeSurvey' },
+					},
+				} );
+			} );
+		} );
+
+		describe( 'sendSurveyEvent', () => {
+			it( 'throws an error when parameters are missing or incorrect', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).sendSurveyEvent();
+				} ).toThrow( 'eventID is required and must be a string' );
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).sendSurveyEvent( 'dismiss_survey', 'answer_question' );
+				} ).toThrow( 'eventData must be an object' );
+			} );
+
+			it( 'does not throw an error when parameters are correct', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).sendSurveyEvent( 'dismiss_survey' );
+				} ).not.toThrow();
+
+				expect( () => {
+					registry.dispatch( STORE_NAME ).sendSurveyEvent( 'answer_question', { foo: 'bar' } );
+				} ).not.toThrow();
+			} );
+
+			it( 'makes network requests to endpoints', async () => {
+				muteFetch( surveyTriggerEndpoint, survey );
+				muteFetch( surveyEventEndpoint, {} );
+
+				// Trigger a survey to appear.
+				await act( () => registry.dispatch( STORE_NAME ).triggerSurvey( 'optimizeSurvey', { ttl: 1 } ) );
+				// Send a survey event.
+				await act( () => registry.dispatch( STORE_NAME ).sendSurveyEvent( 'answer_question', { foo: 'bar' } ) );
```

Comment:
I must have missed this before, but `act` isn't needed outside of component tests.